_includes/specials/taproot/en/09-ptlcs.md

```diff
@@ -0,0 +1,157 @@
+In [last week's column][p4tr sig adaptors], we looked at [signature
+adaptors][topic adaptor signatures] and how the activation of
+[taproot][topic taproot] with [schnorr signatures][topic schnorr
+signatures] will make it easier to use adaptors privately and
+efficiently.  There are several ways signature adaptors can be used on
+Bitcoin but one of the most immediately beneficial will be Point Time
+Lock Contracts ([PTLCs][topic ptlc]), which can replace the
+[venerable][htlc history] Hash Time Lock Contracts ([HTLCs][topic htlc])
+used for years.  This will bring several advantages, but it also comes
+with some challenges.  To understand both, we first start with a
+simplified example of HTLCs in use; the example below could be offchain
+LN payments, onchain coinswaps, or a hybrid onchain/offchain system like
+Lightning Loop---it's this flexibility that makes HTLCs so widely used.
+
+Alice wants to pay Carol by routing a payment through Bob, who neither
+Alice nor Carol wants to trust.  Carol creates a random preimage and hashes
+it with the SHA256 algorithm.  Carol gives the hash to Alice and keeps
+the preimage secret.  Alice initiates a payment
+to Bob which he can claim with a signature for his public key plus the
+preimage; alternatively, Alice can receive a refund after 10 blocks by
+spending the transaction back to herself with a signature for her public
+key.  Here's that [policy][htlc1 minsc] described in the Minsc language:
+
+```hack
+(pk($bob) && sha256($preimage)) || (pk($alice) && older(10))
+```
+
+Bob can now initiate a payment for the same amount (perhaps minus fees) to
+Carol with basically the same script, only with the parties
+[updated][htlc2 minsc] and a lower refund time out.
+
+```hack
+(pk($carol) && sha256($preimage)) || (pk($bob) && older(5))
+```
+
+Now Carol can claim her payment from Bob within five blocks by using the preimage, which
+reveals the preimage to Bob and allows him to claim the payment from
+Alice, also within at least five block.
+
+### Privacy problems with HTLCs
+
+If the scripts above are published onchain, the reuse of the same hash
+and preimage makes it immediately clear that A paid C via
+B.  This can be a significant problem for same-chain and cross-chain
+coinswaps.  Less obviously, this is also a problem for offchain routing
+protocols like LN.  If we imagine a longer routing path where one person
+controls multiple hops along the path, they can see the reuse of
+the same hash and preimage to determine that some nodes are routing
+nodes, increasing the probability that the remaining nodes are either
+the spender or the receiver.   This is one part of the *linkability
+problem* that may be LN's greatest current privacy weakness.
+
+![Illustration of HTLC linkability problem](/img/posts/2021-07-ln-linkability1.dot.png)
+
+Although [multipath payments][topic multipath payments] partially
+mitigates other aspects of LN's linkability problem, e.g. payment amount
```

Comment:
Not positive saying it out loud, but doesn't the plural payments agree with singular the verb mitigate?

```suggestion
mitigate other aspects of LN's linkability problem, e.g. payment amount
```